src/flate2.rs

```diff
@@ -0,0 +1,141 @@
+use core::{
```

Comment:
I think this `flate2` module should be kept private and used by public `deflate`, `zlib` and `gzip` modules. That probably also means taking in just a `Compression` as the public config to set whether or not to use the header correctly.

Reply:
(Obviously just adding a `deflate` module for now).

Reply:
This can wait till later.